test/NodeIPCSMSpec.hs

```diff
@@ -164,27 +193,35 @@ nodeIPCSM serverHandles = StateMachine
   where
     -- | Let's handle just Ping/Pong for now.
     mTransitions :: Model r -> Action r -> Response r -> Model r
-    mTransitions model    action      response        = model
+    mTransitions (RunningModel model) action response   = RunningModel $ model
         { inputMessages     = actionToConcrete action       : inputMessages model
         , outputMessages    = responseToConcreate response  : outputMessages model
         }
+    mTransitions _ ShutdownCmd _                        = ShutdownInit
```

Comment:
If you pick `ShutdownCmd`, it should move the model into `ShutdownInit` which should stop the generation of additional actions/commands?